charts/karmada/templates/post-install-job.yaml

```
@@ -50,6 +50,15 @@ spec:
         - |
           bash <<'EOF'
           set -ex
+
+          # notes: the post-install hook job may be applied before all pods ready who relied on job/secret to be deleted,
+          # actually, thoses pods may be running init-container while main-container haven't be created, so here wait for main-container created.
+          # the sub-command `sed '1d'` is to remove the title line
+          # the sub-command `grep -vE 'Running|Completed'` is to exclude the Running or Completed pods
+          while [[ $(kubectl get pod -n {{ $namespace }} | sed '1d' | grep -vE 'Running|Completed' | wc -l) > 0 ]]; do
```

```suggestion
          while [[ $(kubectl get pods -n {{ $namespace }} --field-selector=status.phase!=Running,status.phase!=Succeeded -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}' | wc -l) > 0 ]]; do
```
Probably more straightforward than sed and grep.

---

nice

---

done and tested ok